app/(main)/league/[leagueId]/entry/[entryId]/week/Week.tsx

```diff
@@ -194,10 +196,12 @@ const Week = ({ entry, league, NFLTeams, week }: IWeekProps): JSX.Element => {
   }, [week, selectedLeague]);
 
   useEffect(() => {
-    getCurrentGameWeek();
-    getUserSelectedTeams();
-    getUserWeeklyPick();
-  }, [user]);
+    if (isSignedIn) {
```

Comment:
This is needed because without authentication the page would be blank?

Reply:
correct.  This ensures the userId is there ... if not then we can't get the leagues the user belongs to.  Basically waiting for the useEffect in AuthContextProvider to run.

Reply:
What happens if they are not signed in?

Reply:
The global spinner spins until they are. 

The authcontext provider handles when getting a user if there is no user session it redirects to /login.

```tsx
 if (!isSessionInLocalStorage()) {
      if (pathname !== '/register') {
        router.push('/login');
      }
      return;
    }
    ```